skills/x-twitter-data-workflow/scripts/plan_x_twitter_workflow.py

```
@@ -0,0 +1,459 @@
+        or parsed.fragment
+        or "\\" in value
+    ):
+        raise ValueError(f"{label} must be a public HTTPS URL without embedded credentials.")
+    hostname = parsed.hostname.casefold().rstrip(".")
+    if hostname == "localhost" or hostname.endswith(".localhost"):
+        raise ValueError(f"{label} must not target a local address.")
+    try:
+        address = ipaddress.ip_address(hostname)
+    except ValueError:
+        if "." not in hostname or all(part.isdigit() for part in hostname.split(".")):
+            raise ValueError(f"{label} must use a public hostname.")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)